cpp/src/tiledb/cxx_api/tiledb_cxx_array_util.cc

```diff
@@ -45,6 +61,10 @@ int ArrayUtil::export_file_to_path(const std::string& file_uri, const std::strin
         uint64_t read_size = result_buffer_elements[FILE_ATTRIBUTE_NAME][1];
         os.write((const char*)data_buffer.data(), read_size);
       }
+      offset += buffer_size;
+      subarray[0] = offset;
+      subarray[1] = (file_size > (offset + buffer_size)) ? (offset + buffer_size -1) : (file_size -1);
+      query.set_subarray(subarray);
```

Comment:
You don't need to set the subarray each time, calling `submit` again will resume the query where it left off.

Reply:
Thanks for your info. I was not certain about that yesterday. Will correct it soon

Reply:
removed those codes to set the subarray. thanks